common/src/main/java/com/wynntils/services/itemrecord/type/SavedItem.java

```diff
@@ -137,5 +151,6 @@ public JsonElement serialize(SavedItem src, Type typeOfSrc, JsonSerializationCon
         }
     }
 
-    private record ItemStackInfo(int itemId, int damage, boolean unbreakable, int color) {}
+    private record ItemStackInfo(
+            String registeredItemName, CustomModelData customModelData, int itemId, int damage, int color) {}
```

Comment:
Is item name as a string the best we can do? I've thought about using item codecs to serialize the whole item. As MC maintains that, it should be okay-ishly future compatible. 

From a quick look at https://docs.fabricmc.net/develop/codecs writing a gson serializer seems simple enough.